File: library-management/typescript/tests/BookBuilder.ts
import { Book } from '../src/Book.js';
import { Isbn } from '../src/Isbn.js';

export class BookBuilder {
  private _title = 'The Pragmatic Programmer';
  private _author = 'Andrew Hunt';
  private _isbn = new Isbn('978-0201616224');
  private _copies = 1;

  titled(title: string): this { this._title = title; return this; }
  by(author: string): this { this._author = author; return this; }
  withIsbn(isbn: string): this { this._isbn = new Isbn(isbn); return this; }
  withCopies(copies: number): this { this._copies = copies; return this; }

  build(): Book { return new Book(this._title, this._author, this._isbn); }

  get copyCount(): number { return this._copies; }
}

File: library-management/python/tests/book_builder.py
from __future__ import annotations

from library_management import Book, Isbn


class BookBuilder:
    def __init__(self) -> None:
        self._title = "The Pragmatic Programmer"
        self._author = "Andrew Hunt"
        self._isbn = Isbn("978-0201616224")
        self._copies = 1

    def titled(self, title: str) -> "BookBuilder":
        self._title = title
        return self

    def by(self, author: str) -> "BookBuilder":
        self._author = author
        return self

    def with_isbn(self, isbn: str) -> "BookBuilder":
        self._isbn = Isbn(isbn)
        return self

    def with_copies(self, copies: int) -> "BookBuilder":
        self._copies = copies
        return self

    def build(self) -> Book:
        return Book(self._title, self._author, self._isbn)

    @property
    def copy_count(self) -> int:
        return self._copies
